teps/0094-improve-task-resources-ux.md

```diff
@@ -0,0 +1,119 @@
+Tekton modifies container resource requirements to comply with any LimitRange policies. For example, if a LimitRange requires a minimum resource
+request, Tekton modifies each container to request at least that minimum.
+See [Presentation: Tekton Resource Requests](https://docs.google.com/presentation/d/1-FNMMbRuxckAInO2aJPtzuINqV-9pr-M25OXQoGbV0s/edit#slide=id.p)
+and [Tekton LimitRange docs](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md#specifying-limitrange-values) for more information.
+
+As a result, all `Step` resource requests other than the largest are ignored by Tekton, causing confusion for users.
+In addition, Tekton doesn't currently account for [ResourceQuotas] when determining container resource requests, as documented in
+[Issue: Resource request not applied to init and sidecar containers](https://github.com/tektoncd/pipeline/issues/2933).
+
+### Inability to Parameterize Resource Requirements
+Pipelines supports [variable replacement](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#using-variable-substitution)
+for several string fields. Non-string fields, or string fields with additional validation, cannot
+currently be parameterized, because values like “$(params.foo)” can’t be unmarshalled from JSON
+into the corresponding Go structs. This problem is discussed in more detail in [Issue: Handling parameter interpolation in fields not designed for it](https://github.com/tektoncd/pipeline/issues/1530).
+In the case of resource requirements, only strings like “100Mi” are accepted by the custom unmarshalling function used for resource
+[Quantities](https://github.com/tektoncd/pipeline/blob/28f950700e99fd22a175eab7e2c803248675cca0/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L89).
+
+### No Support for Specifying Resource Requirements at Runtime
+Compute resource requirements typically depend on runtime constraints, for example:
+
+- Image or code building `Task`s can use different amounts of compute resources
+depending on the image or source being built.
+- Kubeflow pipelines and other data pipelines may have variable resource requirements
+depending on the data being processed.
+- Catalog Tasks should be generally reusable in different environments
+that may have different resource constraints.
+
+The current workaround for lack of this feature is to write a new `Task` for each set of resource constraints, as described in
+[this comment](https://github.com/tektoncd/pipeline/issues/4080#issuecomment-884958486) from a buildah user.
+
+### Goals
+
+- Align `Task` resource requirements API with how `Task`s are scheduled.
+- Allow parameterization of any resource requirement related fields defined in `Task`, including within `Step`s and `Sidecar`s.
+- Add configuration to `TaskRun` allowing users to modify resource requirements defined in the corresponding `Task`,
+including those defined in `Step`s and `Sidecar`s.
```

Comment:
I think this is the reason of this "rescope" but those two item can be tackled completely independently of each other (but we can also group them in this TEP).